### DISCUSSION TITLE
How to set a PIN?

### DISCUSSION THREAD

**Discussion by screendriver** [solokeys/solo2-cli#58] [Q&A]
Today I received my SoloKey v2 and try to figure out how to set a PIN for it. With SoloKey v1 it was a matter of

```sh
$ solo key set-pin
```

How to do this with the `solo2` binary?

**Comment by nickray** [comment]
Using the solo1 CLI should work (or chrome://settings/securityKeys, or whatever other standard method). We don't have a good CTAP library available in Rust yet, hence no CTAP command support in solo2 CLI.

**Comment by maddler** [comment]
was there any update on this? Solo1 doesn't look to be working on newer versions of MacOS.

**Comment by lazytownfan** [comment]
I also recently got my SoloKeys 2 and was trying to find the command to set up a PIN, just like with the SoloKey 1.

However, I found this and there doesn't seem to be a way to do this - until a CTAP library in Rust is implemented into `solo2-cli`.

I recently had to readd my SoloKey 1 to my GitHub accounts, because somehow the SoloKey 1 devices need to be added again so that the PIN can be entered when initializing each of them for the first time with your account.  However, the SoloKeys 2 didn't ask for any PIN.

You can set up a PIN on a SoloKey 2 by using a new SoloKey 2 with no PIN on a service that definitely will ask you for a PIN, such as logging into a personal Microsoft account with no password and only the security key.

**Comment by schdro** [comment]
(Resembles question of closed issue https://github.com/solokeys/solo2-cli/issues/114 )

Given that PIN setting is part of FIDO2/CTAP spec:
Tried with generic FIDO2 tool `fido2-token`?
